src/main/java/org/commcare/suite/model/MenuLoader.java

```diff
@@ -0,0 +1,174 @@
+package org.commcare.suite.model;
+
+import org.commcare.core.process.CommCareInstanceInitializer;
+import org.commcare.modern.session.SessionWrapperInterface;
+import org.commcare.util.CommCarePlatform;
+import org.commcare.util.LoggerInterface;
+import org.javarosa.core.model.condition.EvaluationContext;
+import org.javarosa.core.services.locale.Localization;
+import org.javarosa.xpath.XPathException;
+import org.javarosa.xpath.XPathTypeMismatchException;
+import org.javarosa.xpath.expr.FunctionUtils;
+import org.javarosa.xpath.expr.XPathExpression;
+import org.javarosa.xpath.parser.XPathSyntaxException;
+
+import java.util.Hashtable;
+import java.util.Vector;
+
+/**
+ * Created by willpride on 1/3/17.
+ */
+
+public class MenuLoader {
+
+    private Exception loadException;
+    private String xPathErrorMessage;
+    private MenuDisplayable[] menus;
+    private LoggerInterface loggerInterface;
+
+    public MenuLoader(CommCarePlatform platform,
+                      SessionWrapperInterface sessionWrapper,
+                      String menuId,
+                      LoggerInterface loggerInterface) {
+        this.loggerInterface = loggerInterface;
+        this.getMenuDisplayables(platform, sessionWrapper, menuId);
+    }
+
+    public String getErrorMessage() {
+        if (loadException != null) {
+            String errorMessage = loadException.getMessage();
+            if (loadException instanceof XPathSyntaxException) {
+                errorMessage = Localization.get("app.menu.display.cond.bad.xpath", new String[]{xPathErrorMessage, loadException.getMessage()});
```

Comment:
Hey,

So currently there shouldn't be an assumption that there are any "universal" strings in the localization engine like this, so wee should likely move these to something like loggerInterface.logXPathParseError, loggerInterface.logXPathException() or moving the errors to be non-localized.